### PR TITLE
[ICDS Dashboard]: Back filling LS Launched in the PS and map views

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
##### SUMMARY
https://app.asana.com/0/1179234577945697/1178662132203953/f

### CODE REVIEW

File: custom/icds_reports/management/commands/fix_ls_data.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            SET
+            ls.num_supervisor_launched = ut.num_supervisor_launched
+            FROM (
+                SELECT
+                {update_location},
+                sum(num_supervisor_launched) as num_supervisor_launched
+                FROM "{from_table}"
+                GROUP BY {group_by}, month
+            ) ut
+            WHERE ls.{update_location} = ut.{update_location}
```
Comment (Rohit25negi) [inline custom/icds_reports/management/commands/fix_ls_data.py:78]: good to compare agg_level and month as well here. Or you think that will automatically be taken care by the table names?

**Review by Rohit25negi — APPROVED** [review]
overall looks good. Left two questions

File: custom/icds_reports/management/commands/fix_ls_data.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            self.fix_intial_table(month, next_month_start)
+            self.rollup_query_data(month)
+
+    def _tablename_func(self, agg_level, month):
+        return "{}_{}_{}".format('agg_ls', month.strftime("%Y-%m-%d"), agg_level)
+
+    def rollup_query(self, agg_level, month):
+        locations = ['state_id', 'district_id', 'block_id', 'supervisor_id']
+
+        for i in range(3, agg_level - 1, -1):
```
[the remark this answers is not in the document]
Comment (ShubhamBansal1997) [reply custom/icds_reports/management/commands/fix_ls_data.py:64]: using the array below as a param

File: custom/icds_reports/management/commands/fix_ls_data.py (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/management/commands/fix_ls_data.py:78]: set of queries gonna run for a month
```
CREATE UNLOGGED TABLE "tmp_ls_usage" AS SELECT
    supervisor_id,
    count(*) as form_count
    FROM "ucr_icds-cas_static-ls_usage_forms_85eca43d" ls_usage_ucr
    WHERE timeend<'2017-04-01'
    GROUP BY supervisor_id


UPDATE "agg_ls_2017-04-01_4" agg_ls
    SET num_supervisor_launched = CASE WHEN form_count>0 THEN 1 ELSE 0 END
    from (
        SELECT
            supervisor_id,
            form_count
        FROM tmp_ls_usage
        ) ut
        WHERE agg_ls.supervisor_id = ut.supervisor_id


DROP TABLE "tmp_ls_usage"



            UPDATE "agg_ls_2017-04-01_3" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                block_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_4"
                GROUP BY state_id,district_id,block_id, month
            ) ut
            WHERE ls.block_id = ut.block_id



            UPDATE "agg_ls_2017-04-01_2" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                district_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_3"
                GROUP BY state_id,district_id, month
            ) ut
            WHERE ls.district_id = ut.district_id



            UPDATE "agg_ls_2017-04-01_1" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                state_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_2"
                GROUP BY state_id, month
            ) ut
            WHERE ls.state_id = ut.state_id
```

**Review by Rohit25negi — APPROVED** [review]

**Pull request closed without merging**